babel/dates.py

```diff
@@ -1287,8 +1287,20 @@ def format_period(self, char):
         return get_period_names(locale=self.locale)[period]
 
     def format_frac_seconds(self, num):
-        value = str(self.value.microsecond)
-        return self.format(round(float('.%s' % value), num) * 10**num, num)
+        value = self.value.microsecond / 1000000
+        return self.format(round(value, num) * 10**num, num)
+        """ Returns fractional seconds. Converts microseconds received to the \
+            nearest round-off value according to the given precesion value.
+
+            Earlier the function computes wrong result if the microseconds \
+            received are less than 5 digits.
+
+            d = datetime(2016, 2, 15, 8, 3, 1, 799)
```

Comment:
If you are writing doctests, they should be

```pydoc
>>> foo()
result-of-foo
```
, not
```pydoc
foo()
>>> result-of-foo
```

However these shouldn't probably be here either.